flang/test/Fir/boxproc.fir

```diff
@@ -16,9 +16,7 @@
 
 // CHECK-LABEL: define void @_QPtest_proc_dummy_other(ptr
 // CHECK-SAME:              %[[VAL_0:.*]])
-// CHECK:         %[[VAL_1:.*]] = alloca i32, i64 1, align 4
```

Comment:
Could we accidentally drop alignment requirements when creating a global constant?

Reply:
So, I don't think we have any control over alignment of global values. They are just "aligned as MLIR sees fit", which should work fine for align 4 on i32, but wouldn't work for an align 8 of i32, say. 

Do we even have access to alignment in FIR?  There's no mention in "FIRBuilder.cpp".

Reply:
Looking at it now, it is weird that this is written as `alloca` not `fir.alloca`. Is it from another dialect?

Which would explain how it has alignment, when I can't see a parameter for that in `fir.alloca`

Reply:
@tblah This is an LLVM IR test.
```
// RUN: tco %s | FileCheck %s
```

Reply:
Ahh sorry for the noise. Maybe that test shouldn't be `.fir`, but anyway it wasn't created by this PR.